EngineMath/Vector.cpp

```diff
@@ -8,4 +8,76 @@
 
 namespace EngineMath {
```

Comment:
Everything in this file needs to be moved to the header. Templated function declarations and definitions must be in the same Translation Unit (read: file)